specs/request-handler-plugins.md

```diff
@@ -0,0 +1,116 @@
+# Support a "plug-in" model for request handlers
+
+To allow developers to provide unique handling behaviors for different APIs the Microsoft Graph will have a plug-in model to dynamically load request handlers. This will provide separation of concerns between the software components intercepting network request and those providing responses.
+To enable developers building handlers an abstractions library will be distributed on NuGet.
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | -------- | ----- | --- |
+| 1.0 | 2022-12-19 | Initial specifications | @gavinbarron |
+
+## Configuration file
+
+The existing `appsettings.json` file will be used to provide configuration. Both for determining which request handlers are loaded into the proxy and providing configuration for each of the loaded request handlers. The set of request handlers to be loaded will be specified as an array of objects at the root of the configuration file in the `handlers` property. Each handler must supply a `configSection`, `handerPath`, `name`, and `urlsToWatch` in the handlers array.
+- `configSection` is a string for which there must be a matching object property at the root of the configuration file. If a handler does not provide any configuration options then an empty object must be supplied.
+- `handlerPath` a string which is the relative path to the assembly containing the class definition for the handler which will be loaded using reflection.
+- `name` a string identifier for the handler class, allows developer to ship multiple handlers in a single assembly.
+- `urlsToWatch` is an array of strings defining the urls for which this hander will watch, this behavior is defined in the [Multi URL support spec](./multi-url-support.md).
+
+```json
+  "handlers": [
```

Comment:
Should we add the ability to temporarily disable a plugin without having to remove the whole config block from the config? That would be useful in cases when you want to disable a specific plugin/guidance when testing a specific app.

Reply:
That's a great idea!